Fix javadoc ERRORS

See: https://github.com/eclipse-ee4j/jstl-api/pull/162#issuecomment-960180936

When executing `mvn javadoc:javadoc` I see a number of ERRORS such as the following:

```
:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\sql\ResultImpl.java:32: error: tag not supported in the generated HTML version: tt
[ERROR]  * capable of returing an array of <tt>Row</tt> objects containing a <tt>Column</tt> instance for each column in the
```

I'll use this issue to resolve the ERRORS. There are also a number of WARNINGS that we should look at but they don't prevent a successful build so we can investigate them in more detail in another issue.

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
@@ -34,9 +34,9 @@ import jakarta.servlet.jsp.tagext.TagSupport;
  *
  * <ul>
  * <li>Conditional processing of the action's body based on the returned value of the abstract method
- * <tt>condition()</tt>.</li>
- * <li>Storing the result of <tt>condition()</tt> as a <tt>Boolean</tt> object into a JSP scoped variable identified by
- * attributes <tt>var</tt> and <tt>scope</tt>.
+ * <code>condition()</code>.</li>
+ * <li>Storing the result of <code>condition()</code> as a <code>Boolean</code> object into a JSP scoped variable identified by
+ * attributes <code>var</code> and <code>scope</code>.
  * </ul>
  *
  * @author Shawn Bayern
@@ -48,7 +48,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
     /**
      * <p>
      * Subclasses implement this method to compute the boolean result of the conditional action. This method is invoked once
-     * per tag invocation by <tt>doStartTag()</tt>.
+     * per tag invocation by <code>doStartTag()</code>.
      *
      * @return a boolean representing the condition that a particular subclass uses to drive its conditional logic.
      */
@@ -58,7 +58,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
     // Constructor
 
     /**
-     * Base constructor to initialize local state. As with <tt>TagSupport</tt>, subclasses should not implement constructors
+     * Base constructor to initialize local state. As with <code>TagSupport</code>, subclasses should not implement constructors
      * with arguments, and no-argument constructors implemented by subclasses must call the superclass constructor.
      */
     public ConditionalTagSupport() {
@@ -70,7 +70,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
     // Lifecycle management and implementation of conditional behavior
 
     /**
-     * Includes its body if <tt>condition()</tt> evaluates to true.
+     * Includes its body if <code>condition()</code> evaluates to true.
      */
     @Override
     public int doStartTag() throws JspException {
@@ -111,7 +111,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
     /**
      * Sets the 'var' attribute.
      *
-     * @param var Name of the exported scoped variable storing the result of <tt>condition()</tt>.
+     * @param var Name of the exported scoped variable storing the result of <code>condition()</code>.
      */
     public void setVar(String var) {
         this.var = var;

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/Config.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/Config.java
@@ -84,7 +84,7 @@ public class Config {
      * @param name Configuration variable name
      * @param scope Scope in which the configuration variable is to be looked up
      *
-     * @return The <tt>java.lang.Object</tt> associated with the configuration variable, or null if it is not defined.
+     * @return The <code>java.lang.Object</code> associated with the configuration variable, or null if it is not defined.
      */
     public static Object get(PageContext pageContext, String name, int scope) {
         switch (scope) {
@@ -111,7 +111,7 @@ public class Config {
      * @param request Request object in which the configuration variable is to be looked up
      * @param name Configuration variable name
      *
-     * @return The <tt>java.lang.Object</tt> associated with the configuration variable, or null if it is not defined.
+     * @return The <code>java.lang.Object</code> associated with the configuration variable, or null if it is not defined.
      */
     public static Object get(ServletRequest request, String name) {
         return request.getAttribute(name + REQUEST_SCOPE_SUFFIX);
@@ -128,7 +128,7 @@ public class Config {
      * @param session Session object in which the configuration variable is to be looked up
      * @param name Configuration variable name
      *
-     * @return The <tt>java.lang.Object</tt> associated with the configuration variable, or null if it is not defined, if
+     * @return The <code>java.lang.Object</code> associated with the configuration variable, or null if it is not defined, if
      * session is null, or if the session is invalidated.
      */
     public static Object get(HttpSession session, String name) {
@@ -152,7 +152,7 @@ public class Config {
      * @param context Servlet context in which the configuration variable is to be looked up
      * @param name Configuration variable name
      *
-     * @return The <tt>java.lang.Object</tt> associated with the configuration variable, or null if it is not defined.
+     * @return The <code>java.lang.Object</code> associated with the configuration variable, or null if it is not defined.
      */
     public static Object get(ServletContext context, String name) {
         return context.getAttribute(name + APPLICATION_SCOPE_SUFFIX);
@@ -312,13 +312,13 @@ public class Config {
      *
      * <p>
      * For each of the JSP scopes (page, request, session, application), get the value of the configuration variable
-     * identified by <tt>name</tt> using method <tt>get()</tt>. Return as soon as a non-null value is found. If no value is
-     * found, get the value of the context initialization parameter identified by <tt>name</tt>.
+     * identified by <code>name</code> using method <code>get()</code>. Return as soon as a non-null value is found. If no value is
+     * found, get the value of the context initialization parameter identified by <code>name</code>.
      *
      * @param pageContext Page context in which the configuration setting is to be searched
      * @param name Context initialization parameter name of the configuration setting
      *
-     * @return The <tt>java.lang.Object</tt> associated with the configuration setting identified by <tt>name</tt>, or null
+     * @return The <code>java.lang.Object</code> associated with the configuration setting identified by <code>name</code>, or null
      * if it is not defined.
      */
     public static Object find(PageContext pageContext, String name) {

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTag.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTag.java
@@ -22,13 +22,13 @@ import jakarta.servlet.jsp.tagext.Tag;
 /**
  * <p>
  * The Jakarta Standard Tag Library allows developers to write custom iteration tags by implementing the LoopTag
- * interface. This is not to be confused with <tt>jakarta.servlet.jsp.tagext.IterationTag</tt> as defined in Jakarta
+ * interface. This is not to be confused with <code>jakarta.servlet.jsp.tagext.IterationTag</code> as defined in Jakarta
  * Server Pages 1.2. LoopTag establishes a mechanism for iteration tags to be recognized and for type-safe implicit
  * collaboration with custom subtags.
  *
  * <p>
  * In most cases, it will not be necessary to implement this interface manually, for a base support class
- * (<tt>LoopTagSupport</tt>) is provided to facilitate implementation.
+ * (<code>LoopTagSupport</code>) is provided to facilitate implementation.
  * </p>
  *
  * @author Shawn Bayern
@@ -47,7 +47,7 @@ public interface LoopTag extends Tag {
     /**
      * Retrieves a 'status' object to provide information about the current round of the iteration.
      *
-     * @return The LoopTagStatus for the current <tt>LoopTag</tt>.
+     * @return The LoopTagStatus for the current <code>LoopTag</code>.
      */
     LoopTagStatus getLoopStatus();
 }

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagStatus.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagStatus.java
@@ -70,7 +70,7 @@ public interface LoopTagStatus {
      * 'first' even when getIndex() != 0, for 'index' refers to the absolute index of the current 'item' in the context of
      * its underlying collection. It is always that case that a true result from isFirst() implies getCount() == 1.
      *
-     * @return <tt>true</tt> if the current round is the first in the iteration, <tt>false</tt> otherwise.
+     * @return <code>true</code> if the current round is the first in the iteration, <code>false</code> otherwise.
      */
     boolean isFirst();
 
@@ -79,7 +79,7 @@ public interface LoopTagStatus {
      * is taken into account. isLast() doesn't necessarily refer to the status of the underlying Iterator; it refers to
      * whether or not the current round will be the final round of iteration for the tag associated with this LoopTagStatus.
      *
-     * @return <tt>true</tt> if the current round is the last in the iteration, <tt>false</tt> otherwise.
+     * @return <code>true</code> if the current round is the last in the iteration, <code>false</code> otherwise.
      */
     boolean isLast();
 

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
@@ -41,7 +41,7 @@ import jakarta.servlet.jsp.tagext.TryCatchFinally;
  * <p>
  * Since most iteration tags will behave identically with respect to actual iterative behavior, the Jakarta Standard Tag
  * Library provides this base support class to facilitate implementation. Many iteration tags will extend this and
- * merely implement the <tt>hasNext()</tt> and <tt>next()</tt> methods to provide contents for the handler to iterate
+ * merely implement the <code>hasNext()</code> and <code>next()</code> methods to provide contents for the handler to iterate
  * over.
  * </p>
  *
@@ -50,20 +50,20 @@ import jakarta.servlet.jsp.tagext.TryCatchFinally;
  * </p>
  *
  * <ul>
- * <li>Iteration control, based on protected <tt>prepare()</tt>, <tt>next()</tt>, and <tt>hasNext()</tt> methods
- * <li>Subsetting (<tt>begin</tt>, <tt>end</tt>, <tt>step</tt> functionality, including validation of subset parameters
+ * <li>Iteration control, based on protected <code>prepare()</code>, <code>next()</code>, and <code>hasNext()</code> methods
+ * <li>Subsetting (<code>begin</code>, <code>end</code>, <code>step</code> functionality, including validation of subset parameters
  * for sensibility)
- * <li>item retrieval (<tt>getCurrent()</tt>)
- * <li>status retrieval (<tt>LoopTagStatus</tt>)
- * <li>exposing attributes (set by <tt>var</tt> and <tt>varStatus</tt> attributes)
+ * <li>item retrieval (<code>getCurrent()</code>)
+ * <li>status retrieval (<code>LoopTagStatus</code>)
+ * <li>exposing attributes (set by <code>var</code> and <code>varStatus</code> attributes)
  * </ul>
  *
  * <p>
- * In providing support for these tasks, <tt>LoopTagSupport</tt> contains certain control variables that act to modify
+ * In providing support for these tasks, <code>LoopTagSupport</code> contains certain control variables that act to modify
  * the iteration. Accessors are provided for these control variables when the variables represent information needed or
- * wanted at translation time (e.g., <tt>var</tt>, <tt>varStatus</tt>). For other variables, accessors cannot be
+ * wanted at translation time (e.g., <code>var</code>, <code>varStatus</code>). For other variables, accessors cannot be
  * provided here since subclasses may differ on their implementations of how those accessors are received. For instance,
- * one subclass might accept a <tt>String</tt> and convert it into an object of a specific type by using an expression
+ * one subclass might accept a <code>String</code> and convert it into an object of a specific type by using an expression
  * evaluator; others might accept objects directly. Still others might not want to expose such information to outside
  * control.
  * </p>
@@ -181,7 +181,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
      * See <a href="#next()">next</a> for more information about the purpose and expectations behind this tag.
      * </p>
      *
-     * @return <tt>true</tt> if there is at least one more item to iterate over, <tt>false</tt> otherwise
+     * @return <code>true</code> if there is at least one more item to iterate over, <code>false</code> otherwise
      * @exception jakarta.servlet.jsp.JspTagException
      * @see #next
      */

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/fmt/LocaleSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/fmt/LocaleSupport.java
@@ -51,7 +51,7 @@ public class LocaleSupport {
      *
      * <p>
      * The given key is looked up in the resource bundle of the default I18N localization context, which is retrieved from
-     * the <tt>jakarta.servlet.jsp.jstl.fmt.localizationContext</tt> configuration setting.
+     * the <code>jakarta.servlet.jsp.jstl.fmt.localizationContext</code> configuration setting.
      *
      * <p>
      * If the configuration setting is empty, or the default I18N localization context does not contain any resource bundle,
@@ -89,7 +89,7 @@ public class LocaleSupport {
 
     /**
      * Retrieves the localized message corresponding to the given key, and performs parametric replacement using the
-     * arguments specified via <tt>args</tt>.
+     * arguments specified via <code>args</code>.
      *
      * <p>
      * See the specification of the &lt;fmt:message&gt; action for a description of how parametric replacement is
@@ -111,7 +111,7 @@ public class LocaleSupport {
 
     /**
      * Retrieves the localized message corresponding to the given key, and performs parametric replacement using the
-     * arguments specified via <tt>args</tt>.
+     * arguments specified via <code>args</code>.
      *
      * <p>
      * See the specification of the &lt;fmt:message&gt; action for a description of how parametric replacement is
@@ -187,13 +187,13 @@ public class LocaleSupport {
      * Gets the resource bundle with the given base name, whose locale is determined as follows:
      *
      * Check if a match exists between the ordered set of preferred locales and the available locales, for the given base
-     * name. The set of preferred locales consists of a single locale (if the <tt>jakarta.servlet.jsp.jstl.fmt.locale</tt>
+     * name. The set of preferred locales consists of a single locale (if the <code>jakarta.servlet.jsp.jstl.fmt.locale</code>
      * configuration setting is present) or is equal to the client's preferred locales determined from the client's browser
      * settings.
      *
      * <p>
      * If no match was found in the previous step, check if a match exists between the fallback locale (given by the
-     * <tt>jakarta.servlet.jsp.jstl.fmt.fallbackLocale</tt> configuration setting) and the available locales, for the given
+     * <code>jakarta.servlet.jsp.jstl.fmt.fallbackLocale</code> configuration setting) and the available locales, for the given
      * base name.
      *
      * @param pageContext Page in which the resource bundle with the given base name is requested
@@ -268,7 +268,7 @@ public class LocaleSupport {
      * @param basename the resource bundle's base name
      *
      * @return the localization context containing the resource bundle with the given base name and best matching locale, or
-     * <tt>null</tt> if no resource bundle match was found
+     * <code>null</code> if no resource bundle match was found
      */
     private static LocalizationContext findMatch(PageContext pageContext, String basename) {
         LocalizationContext locCtxt = null;
@@ -297,7 +297,7 @@ public class LocaleSupport {
      *
      * @param pref the preferred locale
      *
-     * @return the requested resource bundle, or <tt>null</tt> if no resource bundle with the given base name exists or if
+     * @return the requested resource bundle, or <code>null</code> if no resource bundle with the given base name exists or if
      * there is no exact- or language-match between the preferred locale and the locale of the bundle returned by
      * java.util.ResourceBundle.getBundle().
      */
@@ -380,14 +380,14 @@ public class LocaleSupport {
 
     /**
      * Parses the given locale string into its language and (optionally) country components, and returns the corresponding
-     * <tt>java.util.Locale</tt> object.
+     * <code>java.util.Locale</code> object.
      *
      * If the given locale string is null or empty, the runtime's default locale is returned.
      *
      * @param locale the locale string to parse
      * @param variant the variant
      *
-     * @return <tt>java.util.Locale</tt> object corresponding to the given locale string, or the runtime's default locale if
+     * @return <code>java.util.Locale</code> object corresponding to the given locale string, or the runtime's default locale if
      * the locale string is null or empty
      *
      * @throws IllegalArgumentException if the given locale does not have a language component or has an empty country

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/sql/Result.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/sql/Result.java
@@ -26,15 +26,15 @@ import java.util.SortedMap;
  * </p>
  *
  * <ul>
- * <li>The result rows (<tt>getRows()</tt> and <tt>getRowsByIndex()</tt>)
- * <li>The column names (<tt>getColumnNames()</tt>)
- * <li>The number of rows in the result (<tt>getRowCount()</tt>)
+ * <li>The result rows (<code>getRows()</code> and <code>getRowsByIndex()</code>)
+ * <li>The column names (<code>getColumnNames()</code>)
+ * <li>The number of rows in the result (<code>getRowCount()</code>)
  * <li>An indication whether the rows returned represent the complete result or just a subset that is limited by a
- * maximum row setting (<tt>isLimitedByMaxRows()</tt>)
+ * maximum row setting (<code>isLimitedByMaxRows()</code>)
  * </ul>
  *
  * <p>
- * An implementation of the <tt>Result</tt> interface provides a <i>disconnected</i> view into the result of a query.
+ * An implementation of the <code>Result</code> interface provides a <i>disconnected</i> view into the result of a query.
  *
  * @author Justyna Horwat
  *
@@ -92,7 +92,7 @@ public interface Result {
     /**
      * Returns true if the query was limited by a maximum row setting
      *
-     * @return <tt>true</tt> if the query was limited by a maximum row setting
+     * @return <code>true</code> if the query was limited by a maximum row setting
      */
     boolean isLimitedByMaxRows();
 }

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/sql/ResultImpl.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/sql/ResultImpl.java
@@ -28,8 +28,8 @@ import java.util.TreeMap;
 
 /**
  * <p>
- * This class creates a cached version of a <tt>ResultSet</tt>. It's represented as a <tt>Result</tt> implementation,
- * capable of returing an array of <tt>Row</tt> objects containing a <tt>Column</tt> instance for each column in the
+ * This class creates a cached version of a <code>ResultSet</code>. It's represented as a <code>Result</code> implementation,
+ * capable of returing an array of <code>Row</code> objects containing a <code>Column</code> instance for each column in the
  * row. It is not part of the Jakarta Standard Tab Library API; it serves merely as a back-end to ResultSupport's static
  * methods. Thus, we scope its access to the package.
  *
@@ -47,7 +47,7 @@ class ResultImpl implements Result, Serializable {
      * This constructor reads the ResultSet and saves a cached copy. It's important to note that this object will be
      * serializable only if the objects returned by the ResultSet are serializable too.
      *
-     * @param resultSet an open <tt>ResultSet</tt>, positioned before the first row
+     * @param resultSet an open <code>ResultSet</code>, positioned before the first row
      * @param startRow beginning row to be cached
      * @param maxRows query maximum rows limit
      * @exception java.sql.SQLException if a database error occurs

